docs/baseband/mark4/index.rst

```diff
@@ -4,48 +4,88 @@
+
 .. _mark4_usage:
 
 Usage
 =====
 
-This section covers Mark 4-specific features of Baseband.  Tutorials for general
+This section covers reading and writing Mark 4 files with Baseband; general
 usage can be found under the :ref:`Using Baseband <using_baseband_toc>` section.
-The examples below use the small sample file ``baseband/data/sample.m4``,
-and assumes the `numpy`, `astropy.units`, and `baseband.mark4` modules have
-been imported::
+The examples below use the small sample file ``baseband/data/sample.m4``, and
+the `numpy`, `astropy.units`, `astropy.time.Time`, and `baseband.mark4`
+modules::
 
     >>> import numpy as np
     >>> import astropy.units as u
+    >>> from astropy.time import Time
     >>> from baseband import mark4
     >>> from baseband.data import SAMPLE_MARK4
 
-Opening a Mark 4 file with :func:`~baseband.mark4.open` in binary mode provides
+Opening a Mark 4 file with `~baseband.mark4.open` in binary mode provides
 a normal file reader but extended with methods to read a
-:class:`~baseband.mark4.Mark4Frame`.  Mark 4 data files generally do not start 
-(or end) at a frame boundary, so in binary mode one has to seek the first frame
-using `~baseband.mark4.base.Mark4StreamReader.find_frame`.  One also has to pass
-in the number of tracks used to `~baseband.mark4.base.Mark4StreamReader.read_frame`,
-and the decade the data were taken, since those numbers cannot be inferred from
-the data themselves::
-
-    >>> fh = mark4.open(SAMPLE_MARK4, 'rb')
-    >>> fh.find_frame(ntrack=64)    # Find first frame.
+`~baseband.mark4.Mark4Frame`.  Mark 4 data files generally **do not
+start (or end) at a frame boundary**, so in binary mode one has to seek the
+first frame using `~baseband.mark4.base.Mark4FileReader.find_frame`.  To use
+`~baseband.mark4.base.Mark4FileReader.find_frame` and 
+`~baseband.mark4.base.Mark4FileReader.read_frame`, one must pass both the
+number of tracks, and either the decade the data was taken, or an equivalent
+reference `~astropy.time.Time` object, since those numbers cannot be inferred
+from the data themselves::
+
+    >>> fb = mark4.open(SAMPLE_MARK4, 'rb')
+    >>> fb.find_frame(ntrack=64)    # Find first frame.
     2696
-    >>> frame = fh.read_frame(ntrack=64, decade=2010)
+    >>> frame = fb.read_frame(ntrack=64, decade=2010)
     >>> frame.shape
     (80000, 8)
-    >>> fh.close()
+    >>> fb.close()
 
 Opening in stream mode automatically seeks for the first frame, and wraps the
 low-level routines such that reading and writing is in units of samples.  It
-also provides access to header information.  In lieu of ``decade``, one may
-also provide a reference time within 4 years of the observation start time::
+also provides access to header information.  Here we pass a reference
+`~astropy.time.Time` object within 4 years of the observation start time to
+``ref_time``, rather than a ``decade``.
+
+::
 
-    >>> from astropy.time import Time
     >>> fh = mark4.open(SAMPLE_MARK4, 'rs', ntrack=64,
```

Comment:
I think we probably should omit `ntrack` here, and write above that the stream reader tries values of `ntrack` until it succeeds.